Source/carthage/Build.swift

```diff
@@ -187,6 +192,66 @@ public struct BuildCommand: CommandType {
 	}
 }
 
+/**
+Returns the SDKs that the given scheme, configuration, and project are capable of building, given
+the signing identities available
+
+:param: sdk           The list of SDKs to filter
+:param: scheme        The scheme name to be built
+:param: configuration The Xcode configuration to be built
+:param: project       The project being built
+:param: formatting    An instance of formatting options
+
+:returns: A list of SDKs that can be built with the configured signing identities
+*/
+public func buildableSDKs(sdks: [SDK], scheme: String, configuration: String, project: ProjectLocator, formatting: ColorOptions.Formatting) -> [SDK] {
+	var identityCheckArgs = BuildArguments(project: project, scheme: scheme, configuration: configuration)
+	
+	var availableIdentities = Set<CodeSigningIdentity>()
+	var result = Set<SDK>()
+	
+	parseSecuritySigningIdentities()
+		|> on(next: {identity in
+			availableIdentities.insert(identity)
+		})
+		|> wait
```

Comment:
You can use `collect` and `first` to return the identities as an array:

```swift
let identities: Result<[CodeSigningIdentity], CarthageError>? = parseCodeSigningIdentities
    |> collect
    |> first
```

It'd probably be better for `buildableSDKs` to take and return a `SignalProducer`, but it should at least only have **1** `wait`. You can do this by combining the identities and SDKs with `combineLatest` and returning the result of `BuildSettings.loadWithArguments` using `flatMap`.